src/rules/nameCasing.test.ts

```diff
@@ -1,44 +1,75 @@
+import { Schema } from "extract-pg-schema";
 import { describe, expect, it, test, vi } from "vitest";
 
+import DeepPartial from "../tests/DeepPartial";
 import { nameCasing } from "./nameCasing";
 
 describe("nameCasing", () => {
   it("no tables or views passed no errors", () => {
     const mockReporter = vi.fn();
+    const schemaObject: DeepPartial<Schema> = {
+      tables: [],
+      views: [],
+    };
 
     nameCasing.process({
       options: [],
-      schemaObject: {
-        tables: [],
-        views: [],
-      },
+      schemaObject: schemaObject as Schema,
       report: mockReporter,
     });
 
     expect(mockReporter).toBeCalledTimes(0);
   });
-  test.each`
-    type             | param       | expected1                        | expected2
-    ${`default`}     | ${null}     | ${`th_is_is_no_sn_a_k_e_ca_s_e`} | ${`neit_he_r_is_this`}
-    ${`snake-case`}  | ${`snake`}  | ${`th_is_is_no_sn_a_k_e_ca_s_e`} | ${`neit_he_r_is_this`}
-    ${`dash case`}   | ${`dash`}   | ${`th-is-is-no-sn-a-k-e-ca-s-e`} | ${`neit-he-r-is-this`}
-    ${`camel case`}  | ${`camel`}  | ${`thIsIsNoSnAKECaSE`}           | ${`neitHeRIsThis`}
-    ${`pascal case`} | ${`pascal`} | ${`ThIsIsNoSnAKECaSE`}           | ${`NeitHeRIsThis`}
-  `(
-    "$type : param of $param applies to table names and requires $expected",
-    ({ _type, param, expected1, expected2 }) => {
+
+  const testCases = [
```

Comment:
All existing tables had the same contents, so I extracted them into one array. The duplication may have been intentional, but arrays require a lot of rows.